### ISSUE TITLE
support karmor tool on windows

### ISSUE THREAD

**Issue by nyrahul** [kubearmor/kubearmor-client#270]
Currently, karmor tool can be installed only from linux. It should be possible to install kubearmor using karmor client tool on windows.

Tasks:
- [ ] Code change to compile on windows
- [ ] karmor client release for windows. Release process change.
- [ ] Installation command change.

**Comment by daemon1024** [comment]
Ref #50

**Comment by akashsawan1** [comment]
Hey @nyrahul ,I would like to take this one on

**Comment by nyrahul** [comment]
> Hey @nyrahul ,I would like to take this one on

Great. Thank you for your interest.

**Comment by akashsawan1** [comment]
Could you help me with some resource to get started.

**Comment by nyrahul** [comment]
> Could you help me with some resource to get started.

KubeArmor uses goreleaser to get the releases prepared for Mac and Linux. You will have to enable windows in there and check if the compilation etc works fine. I guess you will run into issues that will need fixes.

**Comment by khareyash05** [comment]
Hey @nyrahul . Is this issue open to contribute?

**Comment by nyrahul** [comment]
Yes, this is still up as a good-first-issue and we welcome contributions
towards this. Thanks

**Comment by EraKin575** [comment]
@nyrahul is this issue still relevant?

**Comment by nyrahul** [comment]
> @nyrahul is this issue still relevant?

Yes imo it is still relevant. Cc: @daemon1024

**Comment by EraKin575** [comment]
please assign me this issue

**Comment by daemon1024** [comment]
@EraKin575 thanks for taking it up. I see you have asked to be assigned on a couple of issues, let's focus on other issues once we wrap this up. Appreciate the enthusiasm!

**Comment by Manik2708** [comment]
Would like to work on it, please assign @daemon1024 @nyrahul

**Comment by Manik2708** [comment]
@nyrahul So I tried compiling the application for windows. I got errors from following packaged:
1) Probe: This package uses `sys/unix`, so I seperated the logic of printing probe data for windows and unix where unix will print data even if the enviornment is not k8s but in windows if k8s enviornment is not found then it will throw error.
2) VM: This package uses cilium which can't be compiled on windows, so every command related to this package will not be available on windows.
3) Profile: This package uses an accunox package which is using cilium package, so again this command can't be made available for windows.
@nyrahul @daemon1024 Please see if these commands can be removed for windows build. Or are there some alternatives?

**Comment by Manik2708** [comment]
> @nyrahul So I tried compiling the application for windows. I got errors from following packaged:
> 
> 1. Probe: This package uses `sys/unix`, so I seperated the logic of printing probe data for windows and unix where unix will print data even if the enviornment is not k8s but in windows if k8s enviornment is not found then it will throw error.
> 2. VM: This package uses cilium which can't be compiled on windows, so every command related to this package will not be available on windows.
> 3. Profile: This package uses an accunox package which is using cilium package, so again this command can't be made available for windows.
>    @nyrahul @daemon1024 Please see if these commands can be removed for windows build. Or are there some alternatives?
Conclusively it can be said that Karmor in windows will only display data when KubeArmor is running in k8s enviornment and also I think we can make profiling available for windows because cilium is not used in profile, so by picking that code instead of using external package we can make compilation for windows success

**Comment by nyrahul** [comment]
@daemon1024 , I think we should have any dep on Cilium anymore for karmor .. isn't it?
